artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/ActiveMQServerControlImpl.java

```diff
@@ -4479,5 +4488,71 @@ public void replay(String startScan, String endScan, String address, String targ
 
       server.replay(startScanDate, endScanDate, address, target, filter);
    }
+
+   @Override
+   public void stopEmbeddedWebServer() throws Exception {
+      synchronized (embeddedWebServerLock) {
+         getEmbeddedWebServerComponent().stop(true);
+      }
+   }
+
+   @Override
+   public void startEmbeddedWebServer() throws Exception {
+      synchronized (embeddedWebServerLock) {
+         getEmbeddedWebServerComponent().start();
+      }
+   }
+
+   @Override
+   public void restartEmbeddedWebServer() throws Exception {
+      restartEmbeddedWebServer(ActiveMQDefaultConfiguration.getDefaultEmbeddedWebServerRestartTimeout());
+   }
+
+   @Override
+   public void restartEmbeddedWebServer(long timeout) throws Exception {
+      final CountDownLatch latch = new CountDownLatch(1);
+      final AtomicReference<Exception> exception = new AtomicReference<>();
+      /*
+       * This needs to be run in its own thread managed by the broker because if it is run on a thread managed by Jetty
+       * (e.g. if it is invoked from the web console) then the thread will die before Jetty can be restarted.
+       */
+      server.getThreadPool().execute(() -> {
+         try {
+            synchronized (embeddedWebServerLock) {
+               stopEmbeddedWebServer();
+               startEmbeddedWebServer();
+            }
+         } catch (Exception e) {
+            exception.set(e);
+         } finally {
+            latch.countDown();
+         }
+      });
+      boolean result = !latch.await(timeout, TimeUnit.MILLISECONDS);
+      if (result || exception.get() != null) {
+         throw ActiveMQMessageBundle.BUNDLE.embeddedWebServerNotRestarted(timeout, exception.get());
+      }
```

Comment:
```suggestion
      if (!latch.await(timeout, TimeUnit.MILLISECONDS)) {
         throw ActiveMQMessageBundle.BUNDLE.embeddedWebServerRestartTimeout(timeout);
      } else  if (exception.get() != null) {
         throw ActiveMQMessageBundle.BUNDLE.embeddedWebServerRestartFailed(exception.get());
      }
```
Could make the failure mode and exceptions clearer if they were reported separately: either the execution timed out and the state isnt actually known (it may even execute later on just fine), or it failed by throwing while we waited for execution.

(ActiveMQMessageBundle obviously would need changed to match the suggestion change)

Reply:
Yes, that would be more clear.